dnsprovider/pkg/dnsprovider/providers/scaleway/dns.go

```diff
@@ -0,0 +1,571 @@
+package dns
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"golang.org/x/oauth2"
+	"io"
+	"k8s.io/klog/v2"
+	kopsv "k8s.io/kops"
+	"k8s.io/kops/dns-controller/pkg/dns"
+	"k8s.io/kops/dnsprovider/pkg/dnsprovider"
+	"k8s.io/kops/dnsprovider/pkg/dnsprovider/rrstype"
+	"os"
+
+	"github.com/scaleway/scaleway-sdk-go/api/domain/v2beta1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+var _ dnsprovider.Interface = Interface{}
+
+const (
+	ProviderName   = "scaleway"
+	KopsDomainName = "scaleway-terraform.com" // TODO: replace with real domain name
```

Comment:
No hardcoded value, we need to have a way to pass this information from the command line.